public/app/plugins/datasource/graphite/datasource.ts

```diff
@@ -11,6 +11,7 @@ export function GraphiteDatasource(instanceSettings, $q, backendSrv, templateSrv
   this.basicAuth = instanceSettings.basicAuth;
   this.url = instanceSettings.url;
   this.name = instanceSettings.name;
+  this.graphiteVersion = instanceSettings.jsonData.graphiteVersion || '0.9';
```

Comment:
I fixed unit tests, but maybe better to check this instead. When I'm loading DS with old settings, seems, `instanceSettings` contains empty `jsonData` anyway. But not sure how it works in previous grafana versions.